### PR TITLE
[28] Implement Dashboard - part 5

### PR DESCRIPTION
**Pull request by alan-hadyk**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "@testing-library/jest-dom": "^5.1.1",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
+    "character-shuffling": "^1.0.0",
```
Comment (alan-hadyk) [inline package.json]: Remove - unused

File: src/UI/atoms/Text.tsx [hunk 2]
```diff
@@ -21,6 +21,8 @@ function Text({
   paddingTop = "spacing0",
   shouldShuffle = false,
   shouldShuffleOnHover = false,
+  shuffleDelay = 0,
+  shuffleInterval = parseInt(transitionTimes.verySlow),
```
Comment (alan-hadyk) [inline src/UI/atoms/Text.tsx]: Add missing test for `shuffleInterval`

File: src/UI/layout/PositionContainer.tsx [hunk 3]
```diff
@@ -15,6 +15,7 @@ const PositionContainer = ({
   right = "unset",
   top = "unset",
   transform = "none",
+  width = "auto",
```
Comment (alan-hadyk) [inline src/UI/layout/PositionContainer.tsx]: Add missing test for `width`

File: src/UI/layout/SpacingContainer.tsx [hunk 4]
```diff
@@ -18,7 +18,8 @@ const SpacingContainer = ({
   paddingBottom = "spacing0",
   paddingLeft = "spacing0",
   paddingRight = "spacing0",
-  paddingTop = "spacing0"
+  paddingTop = "spacing0",
+  width = "auto"
```
Comment (alan-hadyk) [inline src/UI/layout/SpacingContainer.tsx]: Add missing test for `width`

File: src/UI/molecules/Console.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
+import React, { memo, useRef } from "react";
```
Comment (alan-hadyk) [inline src/UI/molecules/Console.tsx]: Add tests for this component

File: src/UI/molecules/DashboardElement.tsx [hunk 6]
```diff
@@ -34,6 +35,7 @@ function DashboardElement({
         fontSize="font16"
         lineHeight="spacing36"
         shouldShuffle={true}
+        shuffleDelay={shuffleDelay}
```
Comment (alan-hadyk) [inline src/UI/molecules/DashboardElement.tsx]: Add test for `shuffleDelay`

File: src/UI/molecules/FlowChart.tsx [hunk 7]
```diff
@@ -26,15 +27,26 @@ function FlowChart(): JSX.Element {
     > 
       <SpacingContainer
         dataTestId="FlowChartSpacingContainer"
+        height="100%"
```
Comment (alan-hadyk) [inline src/UI/molecules/FlowChart.tsx]: Add test for `height`

File: src/UI/molecules/FlowChart.tsx [hunk 8]
```diff
@@ -26,15 +27,26 @@ function FlowChart(): JSX.Element {
...
         paddingBottom="spacing8"
         paddingLeft="spacing8"
         paddingRight="spacing8"
         paddingTop="spacing8"
+        width="100%"
```
Comment (alan-hadyk) [inline src/UI/molecules/FlowChart.tsx]: Add test for `width`

File: src/UI/molecules/FlowChart.tsx [hunk 9]
```diff
@@ -26,15 +27,26 @@ function FlowChart(): JSX.Element {
...
+        <FlowChart.Icon 
+          id="flow-chart"
+        />
       </SpacingContainer>
     </FlexContainer>
   );
 }
 
+FlowChart.Icon = styled(FluxFlowChart)`
+  display:block;
```
Comment (alan-hadyk) [inline src/UI/molecules/FlowChart.tsx]:
```suggestion
  display: block;
```

File: src/UI/molecules/FlowChart.tsx [hunk 10]
```diff
@@ -26,15 +27,26 @@ function FlowChart(): JSX.Element {
...
-        <FluxFlowChart id="flow-chart" />
+        <FlowChart.Icon 
+          id="flow-chart"
+        />
       </SpacingContainer>
     </FlexContainer>
   );
 }
 
+FlowChart.Icon = styled(FluxFlowChart)`
```
Comment (alan-hadyk) [inline src/UI/molecules/FlowChart.tsx]: Add tests for `FlowChart.Icon`

File: src/UI/organisms/Header.tsx [hunk 11]
```diff
@@ -44,7 +44,7 @@ function Header({
             href={window.location.href}
             iconName="logo"
             height="spacing48"
```
Comment (alan-hadyk) [inline src/UI/organisms/Header.tsx]: Add missing tests for `height` and `width` in all `LinkWithIcon` components

File: src/hooks/__typings__/useShuffleText.d.ts [hunk 12]
```diff
@@ -5,10 +5,12 @@ export interface ShuffleState {
 }
 
 export interface UseShuffleText {
+  duration?: number;
```
Comment (alan-hadyk) [inline src/hooks/__typings__/useShuffleText.d.ts]: Add test for `duration`

File: src/hooks/useShuffleText.ts [hunk 13]
```diff
@@ -17,9 +19,10 @@ export default function useShuffleText({
...
     const shuffle: ShuffleTextType = new ShuffleText(ref.current);
     shuffle.setText(text);
+    shuffle.duration = duration;
 
     onShuffleReady({
-      start: () => shuffle.start()
+      start: () => setTimeout(() => shuffle.start(), shuffleDelay)
     });
-  }, [onShuffleReady, ref, text, shuffleState, shouldInitialize]);
+  }, [duration, onShuffleReady, ref, text, shuffleState, shouldInitialize]);
```
Comment (alan-hadyk) [inline src/hooks/useShuffleText.ts]: Add `shuffleDelay` to dependencies

File: src/pages/Home/sections/dashboard/elements/Code.tsx [hunk 14]
```diff
@@ -5,9 +5,10 @@ import TypingAnimation from "<molecules>/TypingAnimation";
 
 const Code = (): JSX.Element => (
   <DashboardElement
-    childrenHeight="spacing184"
+    childrenHeight="17vh"
     dataTestId="Code"
-    flex="1 1 25%"
+    flex="0 1 25%"
+    shuffleDelay={1800}
```
Comment (alan-hadyk) [inline src/pages/Home/sections/dashboard/elements/Code.tsx]: Add test for `shuffleDelay`

File: src/pages/Home/sections/dashboard/elements/Coords.tsx [hunk 15]
```diff
@@ -5,9 +5,10 @@ import Sonar from "<molecules>/Sonar";
 
 const Coords = (): JSX.Element => (
   <DashboardElement
-    childrenHeight="spacing184"
+    childrenHeight="17vh"
     dataTestId="Coords"
-    flex="1 1 15%"
+    flex="0 1 15%"
+    shuffleDelay={600}
```
Comment (alan-hadyk) [inline src/pages/Home/sections/dashboard/elements/Coords.tsx]: Add test for `shuffleDelay`

File: src/pages/Home/sections/dashboard/elements/Flux.tsx [hunk 16]
```diff
@@ -5,11 +5,12 @@ import FlowChart from "<src>/UI/molecules/FlowChart";
...
 const Flux = (): JSX.Element => (
   <DashboardElement
+    childrenHeight="17vh"
     dataTestId="Flux"
-    flex="1 1 30%"
-    childrenHeight="spacing184"
+    flex="0 1 30%"
     title="Flux" 
     shouldDisplayCorners={true}
+    shuffleDelay={1200}
```
Comment (alan-hadyk) [inline src/pages/Home/sections/dashboard/elements/Flux.tsx]: Add test for `shuffleDelay`

File: src/styles/globalStyle.ts [hunk 17]
```diff
@@ -44,6 +44,7 @@ export default `
     color: ${colorPalette.blue300};
     line-height: 1;
     min-height: 100vh;
+    overflow-x: hidden;
```
Comment (alan-hadyk) [inline src/styles/globalStyle.ts]: Add test for `overflow-x`

File: src/typings/character-shuffling.d.ts [hunk 18]
```diff
@@ -0,0 +1 @@
+declare module "character-shuffling";
```
Comment (alan-hadyk) [inline src/typings/character-shuffling.d.ts]: Remove this file

File: src/hooks/__typings__/useShuffleText.d.ts (hunk 12 above)
Comment (alan-hadyk) [reply src/hooks/__typings__/useShuffleText.d.ts]: It seems that there's no way to test it
